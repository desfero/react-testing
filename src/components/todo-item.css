.todo-item {
    display: flex;
    align-items: center;
}

.todo-item__checkbox {
    --size: 30px;
    width: var(--size);
    height: var(--size);
    margin: 5px;
}

.todo-item__remove {
    margin-left: auto;
}