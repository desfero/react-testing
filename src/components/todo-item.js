import React from 'react';

export const TodoItem = ({item, removeTodo, completeTodo}) => (
  <React.Fragment>
    <input type="checkbox" checked={item.done} onClick={() => completeTodo(item.id, !item.done)} />
    <span>{item.title}</span>
    <button onClick={() => removeTodo(item.id)} className="float-right btn btn-outline-danger">Remove</button>
  </React.Fragment>
);
