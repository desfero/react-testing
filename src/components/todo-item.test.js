import React from 'react';
import {shallow} from 'enzyme';
import {TodoItem} from './todo-item';

describe('TodoItem', () => {

    const todo = {id: 1, title: 'Something'};

    it('should render layout', () => {
        const wrapper = shallow(<TodoItem item={todo}/>);

        expect(wrapper).toMatchSnapshot()
    });

    it('should invoke removeTodo after click on button', () => {
        const removeTodo = jest.fn();

        const wrapper = shallow(<TodoItem item={todo} removeTodo={removeTodo}/>);

        wrapper.find('button').simulate('click');

        expect(removeTodo).toHaveBeenCalledWith(1)
    })
});