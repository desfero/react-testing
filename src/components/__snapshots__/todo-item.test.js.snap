// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem should render layout 1`] = `
<React.Fragment>
  <span>
    Something
  </span>
  <button
    className="float-right btn btn-outline-danger"
    onClick={[Function]}
  >
    Remove
  </button>
</React.Fragment>
`;
